### ISSUE TITLE
Special behaviour for line continuation comment in Vim script

### ISSUE THREAD

**Issue by thinca** [tyru/caw.vim#85]
Since v8.1.0369, Vim script supports "line continuation comment".

```vim
let array = [
"\ this line is a comment
\ 'first element',
"\ line continuation comment must start with `"\ `
\ 'second element',
\ ]
```

Normally, caw.vim makes a line `\ 'text'` into `" \ 'text'` by comment out.  But this is invalid comment.
Can we add a new rule that changes `\ 'text'` into `"\ 'text'` when a line starts with `\`  in Vim script?

**Comment by tyru** [comment]
Thanks for suggestion!
Currently comment string setting is constant.
So I would like to make it variable to specify Funcref, like this.

```vim
" In after/ftplugin/vim/caw.vim
let b:caw_oneline_comment = {lnum -> matchstr(getline(lnum), '^\s*\\') ? '"' : '" '}
```

What do you think about this?

**Comment by thinca** [comment]
It looks very useful!

**Comment by tyru** [comment]
Okay :)
I will implement it!

**Comment by tyru** [comment]
I forgot about `caw_blank_sp` (spaces after comment), not `caw_oneline_comment`.
I implemented this in #87, would you try it?

Settings exists in after/ftplugin/vim/caw.vim already,

```vim
let b:caw_hatpos_sp = {lnum -> getline(lnum) =~# '^\s*\\' ? "" : " "}
let b:caw_zeropos_sp = b:caw_hatpos_sp
```

So line continuation comment should work without configuration.

In line 1, `gcc` results in `" echo {`.
but in line continuation (starting `\`), `gcc` results in `"\ 'foo': 42,` (no spaces after comment)

```vim
echo {
\ 'foo': 42,
\ 'bar': 666,
\}
```

**Comment by tyru** [comment]
Merged.
Please reopen if it does not work.
